cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -0,0 +1,408 @@
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.realtime.kafka;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.templates.etl.api.Emitter;
+import co.cask.cdap.templates.etl.api.realtime.RealtimeContext;
+
+import com.google.common.base.Splitter;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+import com.google.common.cache.RemovalNotification;
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterators;
+
+import kafka.api.FetchRequest;
+import kafka.api.FetchRequestBuilder;
+import kafka.api.PartitionOffsetRequestInfo;
+import kafka.cluster.Broker;
+import kafka.common.ErrorMapping;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.TopicMetadataResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.message.Message;
+import kafka.message.MessageAndOffset;
+
+import org.apache.twill.kafka.client.BrokerInfo;
+import org.apache.twill.kafka.client.BrokerService;
+import org.apache.twill.kafka.client.TopicPartition;
+import org.apache.twill.zookeeper.RetryStrategies;
+import org.apache.twill.zookeeper.ZKClient;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.apache.twill.zookeeper.ZKClientServices;
+import org.apache.twill.zookeeper.ZKClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <p>
+ *  The {@link KafkaSimpleApiConsumer} compatible mode for {@code Kafka} 0.8.x releases.
+ *
+ *  The class expect several runtime arguments:
+ *  -) kafka.zookeeper - the location of Kafka Zookeeper
+ *  -) kafka.brokers - comma separate list of Kafka brokers
+ *  -) kafka.partitions - the number of partitions of the topic
+ *  -) kafka.topic - the Kafka topic to get messages for
+ *  -) kafka.default.offset - the default offset for the partition
+ * </p>
+ */
+public class Kafka08SimpleApiConsumer extends KafkaSimpleApiConsumer<String, ByteBuffer, Long>  {
+  private static final Logger LOG = LoggerFactory.getLogger(Kafka08SimpleApiConsumer.class);
+
+  private ZKClientService zkClient;
+  private BrokerService brokerService;
+  private Cache<TopicPartition, SimpleConsumer> kafkaConsumers;
+
+  @Override
+  protected void configureKafka(KafkaConfigurer configurer) {
+    Map<String, String> runtimeArgs = getContext().getRuntimeArguments();
+    if (runtimeArgs.containsKey("kafka.zookeeper")) {
```

Comment:
Not too familiar with Kafka, but what happens when neither of these keys are present? I see that both these params are not set?